Library/Homebrew/extend/os/linux/dependency_collector.rb

```diff
@@ -0,0 +1,20 @@
+# typed: true
+# frozen_string_literal: true
+
+require "os/linux/glibc"
+
+class DependencyCollector
+  extend T::Sig
+
+  def gcc_dep_if_needed
+    return if DevelopmentTools.gcc_version("gcc") >= OS::LINUX_GCC_CI_VERSION
+
+    add(Dependency.new(CompilerSelector.preferred_gcc, :build))
```

Comment:
@MikeMcQuaid `:build` is wrong here as it gives an error:
```
Error: undefined method `include?' for :build:Symbol
Please report this issue:
  https://docs.brew.sh/Troubleshooting
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dependable.rb:23:in `recommended?'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dependencies_helpers.rb:43:in `block in recursive_includes'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dependency.rb:134:in `block in action'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dependency.rb:132:in `catch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dependency.rb:132:in `action'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dependency.rb:107:in `block in expand'
```

What was your intent here?

Reply:
```suggestion
    add(Dependency.new(CompilerSelector.preferred_gcc, [:build]))
```

will probably fix it. The intent is that this is a build-only dependency. This may not be a correct assumption.